packages/dart_frog/lib/src/request.dart

```diff
@@ -135,6 +135,12 @@ class Request {
     );
   }
 
+  /// Returns the url parameters captured by the [Router].
+  /// Returns an empty map if no parameters are captured.
+  ///
+  /// The returned map is unmodifiable.
```

Comment:
nit: the signature doesn't guarantee is unmodifiable and there doesn't seem to be any test in place to guarantee params is unmodifiable.

Reply:
Yeah good call. I'll adjust -- also want to think a bit about whether we can consolidate this and `mountedParams` which was added a while ago.

Will make some more adjustments later today, thanks for the review!